### DISCUSSION TITLE
tutorials/networking/websocket

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#167] [User-contributed notes]
# tutorials/networking/websocket

HTML5 and WebSocket: The WebSocket protocol was standardized in 2011 with the original goal of allowing browsers to create stable and bidirectional connections with a server. Before that, browsers ...

https://docs.godotengine.org/en/stable/tutorials/networking/websocket.html

**Comment by ReasonPrototype** [comment]
Pls update this, the function: # Our WebSocketServer instance
var _server = WebSocketServer.new()

is no longer working and cant find the new one to use.

**Reply by frozenwaregames** [reply]
This document is really out of date.
Try using this project as an example:
https://github.com/godotengine/godot-demo-projects/tree/master/networking/websocket_chat/websocket

**Reply by GabrielDierks** [reply]
Connecting to wss://echo.websocket.org does not seem to work anymore in with WebSocketPeer.

**Reply by Calinou** [reply]
There's an issue tracking the documentation being outdated:

- https://github.com/godotengine/godot-docs/issues/10031

That said, the service from `echo.websocket.org` is known to have regular downtime, although I can access it via HTTPS right now.

Can you try accessing `wss://echo.websocket.org` using a JavaScript client in a web browser (or another language) to see if it's a Godot-specific issue?

**Reply by Kirill030** [reply]
Hi! Same here.. Alternatively, does WebSocketPeer.new() work with rpc calls? WebSocketServer.new() and WebSocketClient.new() are not being recognised.

**Reply by Calinou** [reply]
WebSocket**Multiplayer**Peer is what works with the [high-level multiplayer](https://docs.godotengine.org/en/stable/tutorials/networking/high_level_multiplayer.html) API, so yes, it'll work with RPCs. For lower-level usage, you'll want to use WebSocketPeer in 4.x instead, which is the class that replaces WebSocketClient and WebSocketServer. It's analogous to PacketPeerUDP for UDP.

**Comment by Poraksis** [comment]
This page informations are incorrect, when i try to connect a external websocket server over golang. It returns header errors. But from browser client i can establish connection. Demo repo mentioned above is working. Someone should update this page at least try to test locally.

**Comment by kkkkkkabc2024** [comment]
Websocketserver.new() can't work now, what's instead?

**Reply by Calinou** [reply]
Pull request opened:

- https://github.com/godotengine/godot-docs/pull/10986
